CesiumGltf/include/CesiumGltf/PropertyTexturePropertyView.h

```diff
@@ -284,173 +419,257 @@ template <typename ElementType> class PropertyTexturePropertyView {
     }
 
-    if constexpr (std::is_same_v<T, int8_t>) {
-      for (size_t i = 0; i < bytes.size(); i++) {
-        result[i] = *reinterpret_cast<const int8_t*>(&bytes[i]);
+    if constexpr (IsMetadataArray<ElementType>::value) {
+      using ArrayElementType = typename MetadataArrayType<ElementType>::type;
+      if constexpr (IsMetadataScalar<ArrayElementType>::value) {
+        return transformNormalizedArray<ArrayElementType>(
+            value,
+            this->offset(),
+            this->scale());
       }
-    }
 
-    if constexpr (std::is_same_v<T, uint8_t>) {
-      for (size_t i = 0; i < bytes.size(); i++) {
-        result[i] = bytes[i];
+      if constexpr (IsMetadataVecN<ArrayElementType>::value) {
+        constexpr glm::length_t N = ArrayElementType::length();
+        using T = typename ArrayElementType::value_type;
+        return transformNormalizedVecNArray<N, T>(
+            value,
+            this->offset(),
+            this->scale());
       }
     }
-
-    return result;
   }
 
-  template <typename T>
-  PropertyArrayView<T>
-  assembleArrayValue(const std::vector<uint8_t>& bytes) const noexcept {
-    std::vector<T> result(bytes.size() / sizeof(T));
-
-    if constexpr (sizeof(T) == 2) {
-      for (int i = 0, b = 0; i < result.size(); i++, b += 2) {
-        using UintType = std::make_unsigned_t<T>;
-        UintType resultAsUint = static_cast<UintType>(bytes[b]) |
-                                (static_cast<UintType>(bytes[b + 1]) << 8);
-        result[i] = *reinterpret_cast<T*>(&resultAsUint);
-      }
-    } else {
-      for (size_t i = 0; i < bytes.size(); i++) {
-        result[i] = *reinterpret_cast<const T*>(&bytes[i]);
-      }
-    }
+  /**
+   * @brief Gets the raw value of the property for the given texture
+   * coordinates. The sampler's wrapping mode will be used when sampling the
+   * texture.
+   *
+   * If this property has a specified "no data" value, the raw value will still
+   * be returned, even if it equals the "no data" value.
+   *
+   * @param u The u-component of the texture coordinates.
+   * @param v The v-component of the texture coordinates.
+   *
+   * @return The value at the nearest pixel to the texture coordinates.
+   */
 
-    return PropertyArrayView<T>(std::move(result));
-  }
+  ElementType getRaw(double u, double v) const noexcept {
+    assert(
+        this->_status == PropertyTexturePropertyViewStatus::Valid &&
+        "Check the status() first to make sure view is valid");
 
-  double applySamplerWrapS(const double u, const int32_t wrapS) const noexcept {
-    if (wrapS == Sampler::WrapS::REPEAT) {
-      double integral = 0;
-      double fraction = std::modf(u, &integral);
-      return fraction < 0 ? 1.0 - fraction : fraction;
-    }
+    double wrappedU = applySamplerWrapS(u, this->_pSampler->wrapS);
+    double wrappedV = applySamplerWrapT(v, this->_pSampler->wrapT);
+
+    // TODO: account for sampler's filter (can be nearest or linear)
```

Comment:
Does this need to be addressed before we can merge? I'm guessing we should at least inform the user that it's not supported, right?

Reply:
We got away with not having it in the previous implementation so I thought it might be ok this round. And I wasn't sure what the official linear (bilinear?) sampling algorithm was, so I ignored it at the time.

I'll see if I can implement it quickly today, to eliminate surprises for the user.

Reply:
Ah, I did have some questions about this... The separation of a min / mag filter makes it possible for one to have a `NEAREST` filter and one to have `LINEAR`. So which one would we use?

I feel like accounting for this setting was awkward because this mainly exists for rendering the texture at different distances. At what point does the user expect the data to be interpolated? Perhaps we could add a parameter to the `get` functions that specifies whether to use linear filtering. It would just have to be user-specified, and I'm not sure how important this functionality is to a user.

I think for now, I'll forgo adding support, and just document that it retrieves data from the nearest pixel.